platform/ios/app-swift/Sources/LocationPrivacyExample.swift

```diff
@@ -9,75 +9,114 @@ enum LocationAccuracyState {
+    private var mapViewModel: MapViewModel
+    // Weak reference because SwiftUI owns the strong reference
+    private(set) weak var mapView: MLNMapView?
+    private var pannedToUserLocation = false
+
+    init(mapViewModel: MapViewModel) {
+        self.mapViewModel = mapViewModel
+        super.init()
+        self.mapViewModel.mapCoordinater = self
+    }
+
+    @MainActor func mapView(_: MLNMapView, didChangeLocationManagerAuthorization manager: MLNLocationManager) {
         guard let accuracySetting = manager.accuracyAuthorization else {
             return
         }
 
         switch accuracySetting() {
         case .fullAccuracy:
-            locationAccuracy = .fullAccuracy
+            mapViewModel.locationAccuracy = .fullAccuracy
         case .reducedAccuracy:
-            locationAccuracy = .reducedAccuracy
+            mapViewModel.locationAccuracy = .reducedAccuracy
         @unknown default:
-            locationAccuracy = .unknown
+            mapViewModel.locationAccuracy = .unknown
         }
     }
 
     // when a location is available for the first time, we fly to it
     func mapView(_ mapView: MLNMapView, didUpdate _: MLNUserLocation?) {
-        if pannedToUserLocation {
-            return
-        }
+        guard !pannedToUserLocation else { return }
         guard let userLocation = mapView.userLocation else {
             print("User location is currently not available.")
             return
         }
         mapView.fly(to: MLNMapCamera(lookingAtCenter: userLocation.coordinate, altitude: 100_000, pitch: 0, heading: 0))
+        pannedToUserLocation = true
+    }
+
+    func mapView(_ mapView: MLNMapView, didFinishLoading _: MLNStyle) {
+        if self.mapView != mapView {
+            self.mapView = mapView
+        }
+    }
+
+    func mapViewDidFinishLoadingMap(_ mapView: MLNMapView) {
+        if self.mapView != mapView {
+            self.mapView = mapView
+        }
+    }
+
+    deinit {
+        // Ensure the coordinator is deallocated if MapLibreViewRepresentable.dismantleUIView was called
     }
 }
 
-struct LocationPrivacyExample: UIViewRepresentable {
-    @Binding var mapView: MLNMapView
-    @Binding var locationAccuracy: LocationAccuracyState
+struct MapLibreViewRepresentable: UIViewRepresentable {
+    let mapViewModel: MapViewModel
 
-    func makeCoordinator() -> Coordinator {
-        Coordinator(mapView: $mapView, locationAccuracy: $locationAccuracy)
+    func makeCoordinator() -> MapLibreRepresentableCoordinator {
+        MapLibreRepresentableCoordinator(mapViewModel: mapViewModel)
     }
 
     func makeUIView(context: Context) -> MLNMapView {
         let mapView = MLNMapView()
-        mapView.showsUserLocation = true
+        // Set delegate first, otherwise delegate callbacks can be missed. For example, setting showsUserLocation calls mapView:didUpdate:userLocation
         mapView.delegate = context.coordinator
-
+        mapView.showsUserLocation = true
```

Comment:
As far as I remember, setting showUserLocation to true immediately triggers a permission request?

Reply:
Yes that is correct. This is something that was already here with @louwers example so I left it alone. I think the example was to just show requesting temporary access. But if we wanted to expand this example we could add two buttons on this example. But I'll let @louwers make that call.

Reply:
If you've tested this and it doesn't clash with that temporary request then fine, I was just worried that this would trigger an auth request on app launch, making your button obsolete...

Reply:
I have tested it, but I see your point. I will double check and I think this will come down to what is in the host apps Info.plist. If the host app by default has `nslocationdefaultaccuracyreduced` set.

https://developer.apple.com/documentation/bundleresources/information_property_list/nslocationdefaultaccuracyreduced

This example was originally created by @louwers , but maybe we should just change the example from requesting temporary precise location access we instead change this example to requesting and showing their location on the map

Let me know @louwers